src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -119,10 +179,36 @@ public void launch() {
             }
         });
 
+
+
+
         kernel.writeEffectiveConfig();
 
         logger.atInfo().setEventType("system-start").addKeyValue("main", kernel.getMain()).log();
         startupAllServices();
+
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private Queue<DeviceIdentityInterface> findProvisioningPlugins() {
+        Queue<DeviceIdentityInterface> provisioningPlugins = new LinkedList<>();
+        EZPlugins ezPlugins = kernel.getContext().get(EZPlugins.class);
+        try {
+            ezPlugins.withCacheDirectory(nucleusPaths.pluginPath());
+            ezPlugins.implementing(DeviceIdentityInterface.class, (c) -> {
```

Comment:
if you have the path to the jar you want to load, then use `ezplugins.loadPlugin` instead of `withCacheDirectory`.

Reply:
We do not have the jar path here in nucleus. Was thinking not to take jar path as kernel args, just taking it as argument to easy setup.